src/mame/misc/toysoldier.cpp

```diff
@@ -0,0 +1,155 @@
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+
+    PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW2:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW2:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW2:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW2:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW2:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW2:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW2:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW2:8")
+INPUT_PORTS_END
+
+void toysoldier_state::program_map(address_map &map)
+{
+	map(0x0000, 0xffff).rom();
+}
+
+void toysoldier_state::io_map(address_map &map)
+{
+	map(0x8000, 0x8001).w("opll", FUNC(ym2413_device::write));
+	map(0x9000, 0x9001).rw("ay", FUNC(ay8910_device::data_r), FUNC(ay8910_device::address_data_w));
+    map(0xa000, 0xa000).portr("IN1"); //  claw up claw down return input
+//	map(0xa000, 0xa000).nopw(); // RL Control
+//  map(0xb000, 0xb000).nopw(); //  aplus ic
+    map(0xb000, 0xb000).portr("IN2"); // joystick
+//	map(0xc000, 0xc000).nopw(); // ???
+    map(0xc000, 0xc000).portr("IN3");  // Coin
+//  map(0xd000, 0xd000).unmapw(); // 7 segment display  control
+    map(0xe000, 0xe000).nopw(); // Super Card
+}
+
+void toysoldier_state::machine_start()
+{
+}
+
+void toysoldier_state::toysoldier(machine_config &config)
+{
+	/* basic machine hardware */
+	i8052_device &maincpu(I80C32(config, "maincpu", XTAL(12'000'000)));
+	maincpu.set_addrmap(AS_PROGRAM, &toysoldier_state::program_map);
+	maincpu.set_addrmap(AS_IO, &toysoldier_state::io_map);
+	/* sound hardware */
+	SPEAKER(config, "mono").front_center();
+	ay8910_device &ay(AY8910(config, "ay", XTAL(12'000'000) / 6));  // Divider not verified
+    ay.add_route(ALL_OUTPUTS, "mono", 1.0);
+	ay.port_a_read_callback().set_ioport("DSW1");
+	ay.port_b_read_callback().set_ioport("DSW2");
+	ym2413_device &opll(YM2413(config, "opll", 3.579545_MHz_XTAL));
+	opll.add_route(ALL_OUTPUTS, "mono", 1.0);
+}
+
+ROM_START( toysoldier )
+	ROM_REGION( 0x10000, "maincpu", 0 )
+	ROM_LOAD( "toysoldier.bin", 0x00000, 0x10000, CRC(BD52C1AE) SHA1(a76c10f93f9087bc2d01bfe866c0e66a006b4ddd) )
+	ROM_REGION( 0x200, "eeprom", 0 ) // according to diagram
+	ROM_LOAD( "93c66.bin", 0x000, 0x200, NO_DUMP )
+	ROM_REGION( 0x40000, "voice", 0 )
+    ROM_LOAD( "api8108a.bin", 0x00000, 0x40000, NO_DUMP ) // api8108A voice rom
+ROM_END
+
+} // anonymous namespace
+
+
+//    YEAR  NAME    PARENT   MACHINE   INPUT   STATE         INIT        ROT   COMPANY      FULLNAME                                                FLAGS
+GAME( 2003, toysoldier, 0,       toysoldier,   toysoldier, toysoldier_state, empty_init, ROT0, "Toy Soldier (Coastal Amusements)", "Toy Soldier", MACHINE_MECHANICAL | MACHINE_NOT_WORKING | MACHINE_IMPERFECT_SOUND ) // This was dumped from an Coastal branded Toy Soldier from around 2003.
```

Comment:
The comment is completely redundant here, as all that information is available in the entries to the GAME macro itself.

Also, COMPANY field should probably be "Coastal Amusements", keeping "Toy Soldier" just in the FULLNAME field.